modules/ui/changeset_editor.js

```diff
@@ -74,10 +74,12 @@ export function uiChangesetEditor(context) {
                 osm.userChangesets(function (err, changesets) {
                     if (err) return;
 
+                    var deleted = (JSON.parse(localStorage.getItem('deletedChangesetComments') || '[]')).map(s => s.trim().toLowerCase());
                     var comments = changesets.map(function(changeset) {
                         var comment = changeset.tags.comment;
                         return comment ? { title: comment, value: comment } : null;
-                    }).filter(Boolean);
+                    }).filter(Boolean)
+                      .filter(c => !deleted.includes(c.value.trim().toLowerCase()));
```

Comment:
The `filter(Boolean)` should always be the last IMO.

Reply:
I think in this case it is ok like this: the `.map(…).filter(Boolean)` is just a shortcut/trick to not have to write `.filter(funct).map(funct')` with two times almost the same function body.